extension/credentialshelper/aws/aws.go

```diff
@@ -79,17 +64,27 @@ func (h *CredentialsHelper) GetCredentials(image *types.TrackedImage) (*types.Cr
 
 	registry := image.Image.Registry()
 
-	if !strings.Contains(registry, "amazonaws.com") {
+	r := regexp.MustCompile(`(?P<registryId>\d+)\.dkr\.ecr\.(?P<region>\S+)\.amazonaws.com`)
+
+	if !r.MatchString(registry) {
 		return nil, credentialshelper.ErrUnsupportedRegistry
 	}
 
+	matches := r.FindStringSubmatch(registry)
+	registryFetched := make(map[string]string)
+	for i, name := range r.SubexpNames() {
```

Comment:
Not sure what's happening here, could you move this into a function and add one or two unit tests with examples?